projects/element-theme/src/styles/bootstrap/forms/_validation.scss

```
@@ -17,6 +17,13 @@
     &.is-#{$state} {
       @content;
     }
+    &.ng-dirty.ng-#{$state} {
+      // show error message when dirty
```

maybe extend the comment here:

```suggestion
      // show error message when dirty, independent of `.ng-touched`
```